docs/rules/replace-j-unit-assert-that-with-hamcrest.md

```diff
@@ -0,0 +1,147 @@
+---
+title: Replace JUnit assertThat with Hamcrest
+ruleId: ReplaceJUnitAssertThatWithHamcrest
+since: 3.29.0
+minJavaVersion: 5
+remediationCost: 2
+    
+description:
+    The JUnit Assert.assertThat method is deprecated. Its sole purpose is to forward the call to the MatcherAssert.assertThat method defined in Hamcrest 1.3. Therefore, it is recommended to directly use the equivalent assertion defined in the third party Hamcrest library.
+tags: ["Java 5", "Testing"]
```

Comment:
As for Tags: Let's discuss whether we shuld add the tag Code Conventions because of the removing of a deprecated method.

Reply:
After discussing it, we will keep it without any tag referring to deprecated code.